src/esp/agent/Agent.h

```diff
@@ -133,14 +133,24 @@ class Agent : public Magnum::SceneGraph::AbstractFeature3D {
 
   /**
    * @brief Return map containing references to superset of all Sensors held by
-   * this Agent's SceneNode and its children values keys are uuid strings,
-   * values are references to Sensors with that uuid
+   * this Agent's SceneNode and its children values.
+   * Keys of map are uuid strings, values are references to Sensors with that
+   * uuid
    */
   std::map<std::string, std::reference_wrapper<sensor::Sensor>>&
   getSubtreeSensors() {
     return node().getSubtreeSensors();
   }
 
+  /**
+   * @brief Return map of uuid keys and Sensor:ptr values of all Sensors held by
+   * this Agent's SceneNode and its children values.
+   * NOTE: This is only called in JS, as emscripten needs pointers
+   */
+  std::map<std::string, sensor::Sensor::ptr> jsGetSubtreeSensors() {
```

Comment:
Let's avoid enlarging this class's public interface unnecessarily. I had in mind a global static function defined in bindings_js.cpp and not exposed to other parts of the codebase. We don't want to litter the C++ codebase with js-specific code.